risc0/zkvm/src/receipt.rs

```diff
@@ -13,6 +13,55 @@
 // limitations under the License.
 
 //! Manages the output and cryptographic data for a proven computation
+//!
+//! # Usage
+//! Receipts are typically constructed as the output of the prover
+//! ```ignore
+//! // In real code, the ELF & Image ID would be generated by risc0 build scripts from guest code
+//! use methods::{EXAMPLE_ELF, EXAMPLE_ID};
+//! use risc0_zkvm::Prover;
+//!
+//! let mut prover = Prover::new(&EXAMPLE_ELF, EXAMPLE_ID)
+//!     .expect("Prover should be constructed from an ELF file and matching Image ID");
```

Comment:
I think example `.expect()` should have messages that pertain to possible ways a function can fail.

Reply:
I think I am going to move the discussion of what leads to `Err` results into the documentation for the relevant functions (so here `Prover::new`) and replace the `expect`s with `?`s. I think that's a more natural place to find such documentation, and anyone who does look for it in these examples and sees the `?` is likely to guess to check the function documentation, whereas the opposite might not be true.

(And then do as you suggest, document ways possible ways for functions that return `Result`s to fail.)